.github/workflows/tests.yml

```diff
@@ -0,0 +1,162 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  # Dockerfile tests
+  docker-build-test:
+    runs-on: ubuntu-latest
+    name: Build docker image
+
+    steps:
+    - uses: actions/checkout@v2
+    - run: make docker-build
+
+  docker-lint:
+    runs-on: ubuntu-latest
+    name: DockerLint
+    steps: 
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: lint
+      uses: brpaz/hadolint-action@master
+
+  # Golang tests
+  go-lint:
+    runs-on: ubuntu-latest
+    name: GoLint
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run golangci-lint
+      uses: actions-contrib/golangci-lint@v1
+      with:
+        golangci_lint_version: 1.22.2
+
+  go-dependencies:
+    runs-on: ubuntu-latest
+    name: Dependencies are up to date
+    steps:
+    - uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+    - name: Checkout
+      uses: actions/checkout@v2
+    - run: go mod tidy
+    - run: go mod vendor
+    - run: go mod graph
+    - run: git status
+    - name: Check diff
+      run: test -z "$(git diff-index --diff-filter=d --name-only HEAD -- 'vendor' | grep -v 'vendor/modules.txt')"
+
+  check-generated-go-files:
+    runs-on: ubuntu-latest
+    name: Generated files are up to date
+    steps:
+    - uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+    - name: Checkout
+      uses: actions/checkout@v2
+    - run: make generate
+    - name: Check diff
+      run: |
+        git restore -- vendor go.mod go.sum
+        git diff-index --exit-code --ignore-cr-at-eol HEAD
+
+  go-tests:
+    runs-on: ubuntu-latest
+    name: K8S v${{ matrix.k8sVersion }} (CM v${{ matrix.certManager }})
+
+    strategy:
+      matrix:
+        # https://github.com/jetstack/cert-manager/branches/active
+        certManager: ["0.12", "0.13", "0.14", "0.15"]
+        # https://snapcraft.io/microk8s
+        k8sVersion: ["1.15", "1.16", "1.17"]
```

Comment:
I think we don't need test so many matrix

Reply:
I had issue with 1.14 when Ingresses were in `extv1` and not yet `netv1`.
IMO it is important to have the compatibility matrix. To be honest we have cluster still running on 1.15...
Moreover it does not cost so much to test on many versions thanks to that matrix syntax.
We could imagine to run 1.15 and 1.16 only on Pull Request? (Not sure how to do that properly)

Reply:
I just feel the checks list is a little long with 43 items.
But it's OK if you really find it would helps.